incipyt/__main__.py

```diff
@@ -76,41 +98,45 @@ def choice_callback(_ctx, _param, _choice):
     default=".venv",
     hidden=True,
 )
-def main(folder, verbose, silent, vcs, env, build, license, **kwargs):  # noqa: A002
+@click.option(
+    "--option",
+    "-o",
+    "options",
+    multiple=True,
+    help="Additionnal key=value pair for the environment, see below",
+)
+def main(folder, verbose, silent, vcs, env, build, check_build, options, **kwargs):
     log_level = DEFAULT_LOGGING_LEVEL - verbose * 10 + silent * 10
     setup_logging(max(logging.NOTSET, min(log_level, logging.CRITICAL)))
 
-    # Populate metadata from other options
-    variables._update_from_dict(**kwargs)
+    try:
+        feed_environ(options, kwargs)
+    except ValueError as exc:
+        raise click.BadArgumentUsage(exc.args[0]) from exc
 
     tools_to_install = [tool for tool in [vcs(), tools.License(), env(), build()] if tool]
-    metadata_setter._stage = "add_to_structure"
 
     if folder == pathlib.Path():
         if any(folder.resolve().iterdir()):
             raise click.BadArgumentUsage(f"FOLDER {folder.resolve()} is not empty.")
-        metadata_setter.default("PROJECT_NAME", folder.resolve().name)
+        project.environ.suggest("PROJECT_NAME", folder.resolve().name)
     else:
         if (folder.is_absolute() and folder.is_dir() and any(folder.iterdir())) or (
             ("." / folder).is_dir() and any(("." / folder).resolve().iterdir())
         ):
             raise click.BadArgumentUsage(f"FOLDER {folder} is not empty.")
-        metadata_setter.default("PROJECT_NAME", folder.name)
-
-    project.environ["LICENSE"] = license
```

Comment:
the license is now part of the `**kwargs` and injected inside the environment cli source via `feed_environ`